docs/blog/2019-05-29-intergrating-graphiql-explorer/index.md

```diff
@@ -0,0 +1,62 @@
+
+<video controls="controls" autoplay="true" loop="true">
+  <source type="video/mp4" src="/graphiql-explore.mp4"></source>
+  <p>Your browser does not support the video element.</p>
+</video>
+
+## What is GraphiQL Explorer?
+
+[`graphiql-explorer`](https://github.com/OneGraph/graphiql-explorer) is a plugin for GraphiQL built by the team at [OneGraph](https://www.onegraph.com/) that adds a new technique to explore and build GraphQL queries. It adds a graphical representation of available fields and inputs that can be used in queries. It also allows constructing full queries by clicking through available fields and inputs without the repetitive process of typing these queries out by hand!
+
+## Why use GraphiQL Explorer?
+
+We often hear that many developers' first usage of GraphQL is through Gatsby. GraphQL, like any technology, has a learning curve. We can't eliminate this learning curve completely but we can try to make it smoother. As mentioned previously, GraphiQL Explorer allows users to build full GraphQL queries without typing a single line of code. This enables users that don't yet fully understand the GraphQL query syntax to _learn_ GraphQL much more easily:
+
+<video controls="controls" autoplay="true" loop="true">
+  <source type="video/mp4" src="./graphiql-explorer-demo.mp4"></source>
+  <p>Your browser does not support the video element.</p>
+</video>
+
+Check out ["How OneGraph onboards users who are new to GraphQL"](https://www.onegraph.com/blog/2019/01/24/How_OneGraph_onboards_users_new_to_GraphQL.html) blog post for more details.
+
+## Advanced usecases
+
+Improvements to on-boarding users new to GraphQL isn't the only goal of integrating GraphiQL Explorer into Gatsby. There are other pain points that are getting addressed with this addition. Specifically [union types and inline fragments](https://graphql.org/learn/queries/#inline-fragments). If the user is not aware of this syntax to query this type of fields it can be a fairly frustrating experience! GraphiQL Explorer helps solve this problem by listing available union types:
+
+<video controls="controls" autoplay="true" loop="true">
+  <source type="video/mp4" src="./graphiql-explorer-union-demo.mp4"></source>
+  <p>Your browser does not support the video element.</p>
+</video>
+
+## Try it now
+
+We recently [added](https://github.com/gatsbyjs/gatsby/pull/14280) GraphiQL Explorer to Gatsby. It's available in starting with `gatsby@2.7.3`.
+
+Create new Gatsby project:
+
+```shell
+gatsby new gatsby-with-explorer
+```
+
+or update Gatsby in your existing project:
+
+```shell
+npm install gatsby
+```
+
+or try it [live](https://gatsby-1774317511.gtsb.io/___graphql?explorerIsOpen=true).
+
+## Future work
+
+There is opportunity for further improvements for Gatsby users. One that we will be tackling next is [support for using GraphQL fragments provided by Gatsby plugins](https://github.com/gatsbyjs/gatsby/issues/14371).
```

Comment:
```suggestion
There are opportunities for further improvements for Gatsby users. One that we will be tackling next is [support for using GraphQL fragments provided by Gatsby plugins](https://github.com/gatsbyjs/gatsby/issues/14371).
```
One other opportunity I'd love to mention is evaluating / improving accessibility of the GraphiQL interface. I haven't gotten a chance to review it really...I could perhaps file an issue to see if anyone wants to take it on?

Reply:
I didn't notice mention about accessibility before. 

I think this is great idea but not sure where the issue should live - we are "just" integrating `graphiql` and `graphiql-explorer` in our code ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-graphiql-explorer/src/app/app.js#L272-L316 ) so evaluating issue/ticket might be filed here, but actual changes (if any required) will need to be done mostly upstream:
 - [`graphiql` repo](https://github.com/graphql/graphiql)
 - [`graphiql-explorer` repo](https://github.com/OneGraph/graphiql-explorer)

Reply:
I was doing quick keyboard navigation check and it doesn't seem great. 

I also found couple of issues already reported in some graphiql related repositories:
 - https://github.com/graphql/graphiql/issues/327
 - https://github.com/codemirror/CodeMirror/issues/4604